src/huggingface_hub/hub_mixin.py

```diff
@@ -248,13 +248,16 @@ def save_pretrained(
         # save model weights/files (framework-specific)
         self._save_pretrained(save_directory)
 
-        # save config (if provided)
+        # save config (if provided and if not serialized yet in `_save_pretrained`)
         if config is None:
             config = self.config
         if config is not None:
             if is_dataclass(config):
                 config = asdict(config)  # type: ignore[arg-type]
-            (save_directory / CONFIG_NAME).write_text(json.dumps(config, sort_keys=True, indent=2))
+            config_path = save_directory / CONFIG_NAME
```

Comment:
This might not work if the user wants to save a file with a name other than the value of the constant `CONFIG_NAME`, e.g. `tokenizer.json`. A library could potentially use a different filename by default, and implement a `filename` kwarg passed in `push_to_hub_kwargs`.
In such case, this can be bypassed by reassigning `CONFIG_NAME` in `save_pretrained`, but there might be a better way to handle this to make sure it doesn't interfere with anything else.

Maybe `ModelHubMixin` could implement a protected `_config_filename` attribute whose value would default to `CONFIG_NAME`, that libraries could freely reassign?

Reply:
Thanks for the concern and suggestion @Natooz!

Actually I would prefer not to allow different config files, at least until we witness a real case in which it would be beneficial. `config.json` is a special filename on the Hub as it is automatically used to count model downloads. If the library author wants 2 configs files -1 handled by the library, 1 handled by huggingface_hub-, then it'd be best if the library-specific config file is named differently (`keras_config.json`, `tokenizer_config.json`, etc.).